### DISCUSSION TITLE
how to use matchComponents()

### DISCUSSION THREAD

**Discussion by mahhmoud-enzyme** [tailwindlabs/tailwindcss#8143] [Help]
In the plugins docs you mentioned a matchComponents() plugin helper to match dynamic styles. but there is now clue on how to use it. 
Can anyone point me to some docs on it.

**Comment by alexanderbluhm** [comment]
@mahhmoud-enzyme Here you can see an example: https://tailwindcss.com/docs/plugins#dynamic-utilities

**Reply by mahhmoud-enzyme** [reply]
I can see that, but I'm asking about the matchComponents method not the matchUtilities 
![Screenshot 2022-04-19 121008](https://user-images.githubusercontent.com/90247981/163981578-07796e49-e220-45ad-b122-350604c7659a.png)

**Reply by mahhmoud-enzyme** [reply]
also, is there a way to have the functionality of the matchUtilities method without having to statically state the values

**Comment by hudri** [comment]
There is an example usage of `matchComponents()` in the aspect ratio plugin:

https://github.com/tailwindlabs/tailwindcss-aspect-ratio/blob/master/src/index.js#L39
